formal-spec/candidateproperties.tex

```diff
@@ -0,0 +1,153 @@
+A software update that has been approved by the stakeholders' community will be eventually applied.
+\end{property}
+
+\begin{property}[Security of Updated Ledger]\label{prop:updated-ledger-security}
+Upon activation of the changes of an approved software update for the updated ledger properties of \emph{persistence} and \emph{liveness} will hold.
+\end{property}
+
+\begin{property}[Conflict Resolution]\label{prop:conflict-res}
+Two or more software updates that have conflicting changes will not be applied both, but a conflict resolution will take place.
+\end{property}
+
+\begin{property}[Update Dependencies]\label{prop:update-dep}
+A software update will never be applied if its dependency requirements are not met
+\end{property}
+
+\begin{property}[Duration of Voting Period]\label{prop:dur-vot-period}
+The period required for approving a software update is not fixed but can be extended by the stakeholders' community.
+\end{property}
+
+\subsection{Properties based on Non-Functional Requirements}
+
+\begin{property}[Transaction Throughput]\label{prop:tr-throughput}
+The ledger transaction throughput is not significantly impacted by the presence of software updates.
+\end{property}
+
+\begin{property}[Blockchain Size]\label{prop:blockchain-size}
+The blockchain size is not significantly impacted by the presence of software updates.
+\end{property}
+
+\subsection{Consensus Protocol Properties}
+
+\begin{property}[Bootstrapping]\label{prop:bootstrap}
+The update mechanism leverages the consensus protocol of the blockchain itself.
+\end{property}
+
+\begin{property}[Resiliency]\label{prop:resiliency}
+The resiliency of the consensus protocol (expressed as the fraction ($t/n$ where $t$ is the adversary stake and $n$ is the total stake) of misbehaving parties a protocol can tolerate) is not affected by the update mechanism
+\end{property}
+
+\begin{property}[Running Time]\label{prop:run-time}
+The worst number of rounds by which honest parties terminate the consensus protocol (i.e., reach a consensus) is not affected.
+\end{property}
+
+\begin{property}[Communication Complexity]\label{prop:comm-complex}
+The worst total number of bits/messages communicated during a consensus protocol run is not affected.
+\end{property}
+
+\subsection{Ideation Phase Properties}
+
+\begin{property}[Uniqueness of SIPs]\label{prop:unique-sip}
+Submitted SIPs in the trace are unique.
+\end{property}
+
+\begin{property}[Valid Ownership of SIPs]\label{prop:owner-sip}
+SIPs can be submitted only by stakeholders of the ledger.
+\end{property}
+
+\begin{property}[Validity of Revealed SIPs]\label{prop:reveal-valid}
+Every Revealed SIP must correspond to one and only stable submitted SIP.
+\end{property}
+
+\begin{property}[Validity of a SIP]\label{prop:sip-valid}
+Every valid SIP upon revealing must comprise a valid set of metadata and an active URL pointing to an off-chain stored SIP bundle, whose structure conforms to the SIP structure guideline.
```

Comment:
> valid set of metadata and an active URL

I don't think we can prove or test that the URL is valid. And the check might not even make sense. Even if the URL points to an online resource, it can become offline later on. 

I don't know if we can do much about this...